collector/lib/Logging.h

```diff
@@ -55,24 +55,45 @@ bool ParseLogLevelName(std::string name, LogLevel* level);
 const char* GetGlobalLogPrefix();
 void SetGlobalLogPrefix(const char* prefix);
 
+const size_t LevelPaddingWidth = 7;
+
 class LogMessage {
  public:
-  LogMessage(const char* file, int line, LogLevel level)
-      : file_(file), line_(line), level_(level) {}
+  LogMessage(const char* file, int line, bool throttled, LogLevel level)
+      : file_(file), line_(line), level_(level), throttled_(throttled) {
+    // if in debug mode, output file names associated with log messages
+    include_file_ = CheckLogLevel(LogLevel::DEBUG);
+  }
 
   ~LogMessage() {
-    const char* basename = strrchr(file_, '/');
-    if (!basename) {
-      basename = file_;
-    } else {
-      ++basename;
-    }
     auto now = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
     auto nowTm = gmtime(&now);
-    std::cerr << GetGlobalLogPrefix()
-              << "[" << GetLogLevelShortName(level_)
-              << " " << std::put_time(nowTm, "%Y%m%d %H%M%S")
-              << " " << basename << ":" << line_ << "] " << buf_.str() << std::endl;
+
+    std::stringstream header;
```

Comment:
I don't know if we want to create a `stringstream` every time we log data, that seems like a thing that could potentially become quite expensive. Why can't we log directly to `std::cerr` like we used to?

Reply:
I'm not sure how much extra overhead this is (since we're storing all parts of the log message in a string stream as well)

But this is just left over from when I was padding the logs based on the width of the whole header instead of just the log level so I'll move it back to writing straight to the stream